lib/HLSL/DxilAddPixelHitInstrumentation.cpp

```diff
@@ -200,15 +208,12 @@ bool DxilAddPixelHitInstrumentation::runOnModule(Module &M)
         }
 
         // Step 2: Calculate pixel index
-        Value * ClampedIndex;
+        Value * Index;
         {
           Constant* RTWidthArg = HlslOP->GetI32Const(RTWidth);
-          auto YOffset = Builder.CreateMul(YAsInt, RTWidthArg);
-          auto Index = Builder.CreateAdd(XAsInt, YOffset);
-
-          // Step 3: Clamp to size of UAV to prevent TDR if something goes wrong
-          auto CompareToLimit = Builder.CreateICmpUGT(Index, NumPixelsMinusOneArg);
-          ClampedIndex = Builder.CreateSelect(CompareToLimit, NumPixelsMinusOneArg, Index, "Clamped");
+          auto YOffset = Builder.CreateMul(YAsInt, RTWidthArg, "YOffset");
+          auto Elementoffset = Builder.CreateAdd(XAsInt, YOffset, "ElementOffset");
+          Index = Builder.CreateMul(Elementoffset, HlslOP->GetU32Const(4), "ByteIndex");
```

Comment:
Why the clamp code is removed?

Reply:
Because it's unnecessary: Only rasterized pixels run this instrumentation, and they will have SV_Position within the expected range.